compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -1188,7 +1188,9 @@ static void addLowerToLLVMGPUPasses(OpPassManager &modulePassManager,
       .addPass(createLLVMGPUVectorLoweringPass)
       .addPass(createExpandGPUOpsPass)
       // Barrier elimination before we reach unstructured control flow.
-      .addPass(createGpuEliminateBarriers);
+      .addPass(createGpuEliminateBarriers)
+      .addPass(createCanonicalizerPass)
```

Comment:
Are these additional canonicalization passes here needed in this PR? Looks like the new pattern is run in `createLLVMGPUVectorLoweringPass`

Reply:
Not really, but it is still good to have them to connect them to amdgpu-maskedload-to-load pass. The reason for really adding adding this canonicalization is to connect it to that pass properly, which is why these were added.